packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -1008,6 +1021,7 @@ export async function estimateAccountUpgradeFee(
       contractAddress,
       privateKey,
       txnInvocation,
+      constants.TRANSACTION_VERSION.V2,
```

Comment:
AccountUpgradeFee will always require V2. While TRANSACTION_VERSION is meant to be the default transaction used, which could be potentially switch to V3 by default.

Reply:
i think if we consider that, CAIRO_VERSION_LEGACY has to change as well

i suppose this method should be reused for every upgrade 
therefore instead of hardcode, using TRANSACTION_VERSION seem better?